Allow showing certain themes at the beginning of a category list

markbiek opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Fixes #100015

## Proposed Changes

Allows us to prioritize a particular theme within in a category.

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

We want to make sure the Course theme is listed first when the Education category is selected.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Run locally or use the calypso.live url
* Go through `/start` until you reach the `designSetup step.
* The "Course" theme should appear first in the list.

<img width="1438" alt="CleanShot 2025-02-25 at 14 04 38@2x" src="https://github.com/user-attachments/assets/93a37f6a-f4ce-4a7a-86d7-82c9098dafd5" />


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

markbiek committed:
Allow showing certain themes at the beginning of a category list

Currently used to prioritze the Course theme in the education category

gabrielcaires commented on packages/design-picker/src/components/unified-design-picker.tsx:
```diff
@@ -349,6 +349,7 @@ interface DesignPickerProps {
 	showActiveThemeBadge?: boolean;
 	isMultiFilterEnabled?: boolean;
 	isBigSkyEligible?: boolean;
+	isIntentCreateCourseGoalEnabled?: boolean;
```
What do you think about passing the `[ 'education' ] = 'course'; directly as a prop instead of getting the feature flag outside and drilling down the value to the component? 

It will make the component prioritize configurable themes by category and reduce the coupling of the feature flag state.

markbiek replied:
That's a good idea! I can do that when I'm removing the feature flag code.

markbiek committed:
Refactor priority themes

First, we remove all of the feature flag code since we've decided to roll this out immediately.

Second, we've simplified how the priority themes are passed down

sixhours commented:
I think this is working but I had to use slightly different testing instructions. I clicked on the Education tab in the `designSetup` step to see the Course theme at the front. And I removed the flag with `&flags=-onboarding/create-course` in the last testing step to see it moved back to the end.

markbiek commented:
> I think this is working but I had to use slightly different testing instructions. I clicked on the Education tab in the `designSetup` step to see the Course theme at the front. And I removed the flag with `&flags=-onboarding/create-course` in the last testing step to see it moved back to the end.

@sixhours I actually just pushed a refactor based on Gabriel's suggestion above and the testing instructions have changed slightly. No more feature flag!
